### PR TITLE
Feature/find update

### LINKED ISSUE #165: Fix FindCommand to with new filtering logic

### PR DESCRIPTION
**Pull request by jinhanjinhan**
Closes #165

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -37,28 +41,41 @@ public class FindCommand extends Command {
...
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(this.combinedPredicate);
+
+        // PredicateAdapter converts Predicate<Participation> to Predicate<Person>
+        Predicate<Person> predicateForPersonFromParticipation = new PredicateAdapter(
+                participationPredicates,
+                model.getParticipationList());
+        this.personPredicates.add(predicateForPersonFromParticipation);
```
Comment (gladwintan) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:75]: Nice implementation to workaround the participation list

**Review by gladwintan — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -37,28 +41,41 @@ public class FindCommand extends Command {
...
+     *
+     * <p>This command uses the provided {@code personPredicates} to filter {@code Person} objects
+     * and the {@code participationPredicates} to filter {@code Participation} records related with those persons.</p>
      *
-     * @param predicates The list of predicates to apply to the search.
-     * @throws NullPointerException If the provided list of predicates is {@code null}.
+     * @param personPredicates A list of {@code Predicate<Person>} for filtering persons based on specific conditions.
+     *                         Must not be {@code null}.
+     * @param participationPredicates A list of {@code Predicate<Participation>} for filtering participation records
+     *                                associated with each person. Must not be {@code null}.
```
Comment (Slipperybara) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:57]: Are the indentation levels correct here?

**Review by Slipperybara — APPROVED** [review]
Looks good

**Pull request merged**